Src/Model/Adapter/Adapter.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Aot\Model\Adapter;
+
+abstract class Adapter
```

Comment:
You must have an interface on top of the abstract class. It allows to implement the interface without the logic from abstract. This implementation is specific to file